### ISSUE TITLE
build fails around librfxencode on CentOS 6

### ISSUE THREAD

**Issue by metalefty** [neutrinolabs/xrdp#693]
Found by X11RDP-RH-Matic travis build: https://travis-ci.org/metalefty/X11RDP-RH-Matic/jobs/195433217

On CentOS 6, build fails with this error message.
```
  CC     xrdp_wm.o
make[2]: *** No rule to make target `../librfxcodec/src/.libs/librfxencode.a', needed by `xrdp'.  Stop.
make[2]: Leaving directory `/home/travis/rpmbuild/BUILD/neutrinolabs-xrdp-8be8347/xrdp'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/travis/rpmbuild/BUILD/neutrinolabs-xrdp-8be8347'
make: *** [all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.ZEIph0 (%build)
```

I'm not sure the diffrence between `.la` and `.a` but this Debian patch fixed the build.
http://sources.debian.net/src/xrdp/0.9.1-7/debian/patches/make-fixes.diff/#L38

```diff
--- a/xrdp/Makefile.am
+++ b/xrdp/Makefile.am
@@ -24,7 +24,7 @@ endif
 if XRDP_RFXCODEC
 AM_CPPFLAGS += -DXRDP_RFXCODEC
 AM_CPPFLAGS += -I$(top_srcdir)/librfxcodec/include
-XRDP_EXTRA_LIBS += $(top_builddir)/librfxcodec/src/.libs/librfxencode.a
+XRDP_EXTRA_LIBS += $(top_builddir)/librfxcodec/src/.libs/librfxencode.la
 endif
 
 if XRDP_PIXMAN
```

**Comment by proski** [comment]
I assume CentOS is not building the static library. I think we should consider using the shared library. But then we should fix PIC support in x86 assembly first. Until then, it's better to enable static libraries.

**Comment by metalefty** [comment]
BTW, CentOS 7 generates `librfxencode.a`.  Does the Debian patch have any side effects?

**Comment by proski** [comment]
The side effect is that xrdp is linked against librfxcodec.so. It is a good thing, but only after we fix the assembly code to be position independent.

**Comment by jerryhall** [comment]
My build for xrdp is effectively moot until this is fixed? I hang on make with this error. I get:

`
Making all in librfxcodec
make[2]: Entering directory `/root/git/neutrinolabs/xrdp/librfxcodec'
make[2]: *** No rule to make target `all'.  Stop.
make[2]: Leaving directory `/root/git/neutrinolabs/xrdp/librfxcodec'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/root/git/neutrinolabs/xrdp'
make: *** [all] Error 2
`

**Comment by metalefty** [comment]
Now out of scope.
